modules/account-lib/src/coin/sol/utils.ts

```diff
@@ -1,34 +1,76 @@
-import { BaseUtils } from '../baseCoin';
-import { NotImplementedError } from '../baseCoin/errors';
+import { Keypair, PublicKey } from '@solana/web3.js';
+import bs58 from 'bs58';
+import BigNumber from 'bignumber.js';
 
-export class Utils implements BaseUtils {
-  /** @inheritdoc */
-  isValidAddress(address: string): boolean {
-    throw new NotImplementedError('isValidAddress not implemented');
-  }
+const DECODED_BLOCK_HASH_LENGTH = 32; // https://docs.solana.com/developing/programming-model/transactions#blockhash-format
+const DECODED_ADDRESS_LENGTH = 32; // https://docs.solana.com/developing/programming-model/transactions#account-address-format
+const DECODED_SIGNATURE_LENGTH = 64; // https://docs.solana.com/terminology#signature
+const BASE_58_ENCONDING_REGEX = '[1-9A-HJ-NP-Za-km-z]';
 
-  /** @inheritdoc */
-  isValidBlockId(hash: string): boolean {
-    throw new NotImplementedError('isValidBlockId not implemented');
-  }
+/** @inheritdoc */
+export function isValidAddress(address: string): boolean {
+  return isValidPublicKey(address);
+}
 
-  /** @inheritdoc */
-  isValidPrivateKey(key: string): boolean {
-    throw new NotImplementedError('isValidPrivateKey not implemented');
-  }
+/** @inheritdoc */
+export function isValidBlockId(hash: string): boolean {
+  return (
+    !!hash &&
+    new RegExp(BASE_58_ENCONDING_REGEX + '{32,44}').test(hash) &&
+    bs58.decode(hash).length === DECODED_BLOCK_HASH_LENGTH
+  );
+}
 
-  /** @inheritdoc */
-  isValidPublicKey(key: string): boolean {
-    throw new NotImplementedError('isValidPublicKey not implemented');
+/** @inheritdoc */
+export function isValidPrivateKey(prvKey: string | Uint8Array): boolean {
+  const key: Uint8Array = typeof prvKey === 'string' ? this.base58ToUint8Array(prvKey) : prvKey;
+  try {
+    return !!Keypair.fromSecretKey(key);
+  } catch (e) {
+    return false;
   }
+}
 
-  /** @inheritdoc */
-  isValidSignature(signature: string): boolean {
-    throw new NotImplementedError('isValidSignature not implemented');
+/** @inheritdoc */
+export function isValidPublicKey(pubKey: string): boolean {
+  try {
+    return (
+      !!pubKey &&
+      new RegExp(BASE_58_ENCONDING_REGEX + '{32,44}').test(pubKey) &&
```

Comment:
Should check here max length.

Reply:
new RegExp(BASE_58_ENCONDING_REGEX + '{32,44}').test(pubKey)
may be '{32,44} should be optional or not. Need further tests.

Reply:
Removed the length check from regex since is speculative.